src/app/components/detail-panel/detail-panel.component.ts

```diff
@@ -1,23 +1,42 @@
-import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
+import { Component, OnInit, ChangeDetectionStrategy, Input, OnDestroy } from '@angular/core';
 import { Element, ElementType, MainService } from 'src/app/services/main.service';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
-  selector: 'ui-detail-panel',
+  selector: 'app-ui-detail-panel',
   templateUrl: './detail-panel.component.html',
   styleUrls: ['./detail-panel.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class DetailPanelComponent implements OnInit {
+export class DetailPanelComponent implements OnInit, OnDestroy {
+  private readonly toDestroy$ = new Subject();
   @Input()
   public element: Element;
 
   public elementTypes: Array<ElementType>;
 
   constructor(private readonly mainService: MainService) {}
 
-  ngOnInit() {
-    (async () => {
-      this.elementTypes = await this.mainService.getAllElementTypes().toPromise();
-    })();
+  ngOnInit(): void {
+    this.initElementTypes();
+  }
+
+  private initElementTypes = (): void => {
```

Comment:
I move this to a separate function and chained with destroy-able token so when component get disposed in the middle of async task that not yet completed, it'll cancel the process and releases the mem.

And, including the error handling as well as avoiding the use of `try catch` block.